lib/Rex/CLI.pm

```diff
@@ -455,6 +454,30 @@ sub __help__ {
 
 }
 
+# Check whether Rex/Template.pm
+# or Rex/Template/NG.pm modules are loaded. If there are,
+# they should be present in the %INC hash.
+sub check_template_modules {
+  my @tpl_modules_loaded =
+    grep { /(Rex\/Template|Rex\/Template\/NG)/ } keys %INC;
+  if ( scalar @tpl_modules_loaded > 0 ) {
+    Rex::Logger::info(
+      "WARNING! You are using Rex built-in template engine aka Rex::Template.",
+      "warn"
+    );
+    Rex::Logger::info(
+      "This engine is obsolete, unmaintained and will be retired in future releases.",
+      "warn"
+    );
```

Comment:
That would also show up when internal core modules import Rex::Template and the user doesn't use it directly. Is there a good way to make an exception based on who imports it?

Reply:
Good point but I don't really know how to solve this problem though.